resources/js/shared/components/flights/FlightsTable.tsx

```diff
@@ -0,0 +1,254 @@
+import { ChangeEvent, useState } from "react";
+import { useMutation, useQuery, useQueryClient } from "@tanstack/react-query";
+
+import {
+  Flight,
+  FlightFilters,
+  ServicePagination,
+  SortDirection,
+  SortDirectionType,
+} from "~/api/api.types";
+import { getCities } from "~/api/cities";
+import { deleteFlight, getFlights } from "~/api/flights";
+import { CreateFlightModal } from "~/modals/Flight/CreateFlightModal";
+import { EditFlightModal } from "~/modals/Flight/EditFlightModal";
+import { Button, errorToast, useToastStore } from "~/ui";
+
+type FilterName = keyof FlightFilters;
+
+export const FlightsTable = () => {
+  const queryClient = useQueryClient();
+  const { pushToast } = useToastStore();
+
+  const [sort, setSort] = useState<string | null>(null);
+  const [sortOrder, setSortOrder] = useState<SortDirectionType>(
+    SortDirection.asc,
+  );
+  const [filters, setFilters] = useState<FlightFilters>({});
+  const [pagination, setPagination] = useState<ServicePagination | null>(null);
```

Comment:
This way typescript infer that pagination could be undefined or ServicePagination

```suggestion
  const [pagination, setPagination] = useState<ServicePagination>();
```